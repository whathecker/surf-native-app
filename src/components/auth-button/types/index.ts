export type AuthButtonProps = {
  authBrand: "apple" | "facebook" | "google";
  screenType: "signin" | "signup";
};
