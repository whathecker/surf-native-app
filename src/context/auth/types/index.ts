export type AuthState = {
  token: string | null;
  errorMsg?: string;
};

export type AuthContextProviderProps = {
  state: AuthState;
  signIn: (authIdp: string) => Promise<void>;
  restoreToken: () => Promise<void>;
  signOut: () => Promise<void>;
};

export enum AuthActionType {
  restore = "RESTORE_TOKEN",
  signIn = "SIGN_IN",
  signOut = "SIGN_OUT",
  error = "ERROR",
}

export type AuthAction = {
  type: AuthActionType;
  payload?: AuthState;
};
