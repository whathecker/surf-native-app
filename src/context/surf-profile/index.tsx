export {
  SurfProfileContext,
  SurfProfileContextProvider,
} from "./surf-profile.context";
