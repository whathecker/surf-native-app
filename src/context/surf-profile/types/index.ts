import {
  SurfProfile,
  SurfLevelQuestionsHolder,
} from "../../../types/surf-profile";

export type SurfProfileState = {
  surfProfile?: SurfProfile;
  errorMsg?: string;
};

export type ContextProviderProps = {
  state: SurfProfileState;
  fetchSurfProfile: () => Promise<void>;
  createSurfProfile: (payload: SurfLevelQuestionsHolder) => Promise<void>;
};

export enum SurfProfileActionType {
  fetch = "FETCH_PROFILE",
  error = "ERROR",
}

export type SurfProfileAction = {
  type: SurfProfileActionType;
  payload?: SurfProfileState;
};
