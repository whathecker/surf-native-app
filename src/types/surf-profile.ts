export type SurfProfile = {
  surfLevelScore: number | null;
};
