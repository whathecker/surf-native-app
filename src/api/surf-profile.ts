import { axiosSurf } from "./instances";
import { SurfProfile, SurfLevelQuestionsHolder } from "../types/surf-profile";
import { secureStorage, getApiError } from "../utils";

const getSurfProfile = async (
  expectedResult: boolean,
): Promise<SurfProfile> => {
  try {
    const token = await secureStorage.getValue("token");

    await axiosSurf.get("/test", {
      headers: { Authorization: `Bearer ${token}` },
    });

    if (expectedResult) {
      return Promise.resolve({ surfLevelScore: 50 });
    } else {
      return Promise.resolve({ surfLevelScore: null });
    }
  } catch (error) {
    const apiError = getApiError(error.message);
    return Promise.reject(apiError);
  }
};

const postSurfProfile = async ({
  selectedSurfLevel,
}: SurfLevelQuestionsHolder): Promise<SurfProfile> => {
  try {
    // Send selectedSurfLevel and questions to backend
    // Receive SurfProfile with surflevelScore as result

    if (!selectedSurfLevel) {
      // What do I do here? throw error?
      throw new Error("Select surf level before sending data");
    }
    return Promise.resolve({ surfLevelScore: 50 });
  } catch (error) {
    const apiError = getApiError(error.message);
    return Promise.reject(apiError);
  }
};

export default { getSurfProfile, postSurfProfile };
