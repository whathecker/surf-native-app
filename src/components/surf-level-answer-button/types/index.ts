import { SurfLevelAnswerOption } from "../../../types/surf-profile";

export type SurfLevelAnswerButtonProps = {
  answerOption: SurfLevelAnswerOption;
};
