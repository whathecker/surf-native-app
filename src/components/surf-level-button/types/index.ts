import { SurfLevel, SelectedSurfLevel } from "../../../types/surf-profile";

export type SetSelectedLevelFunc = (input: SurfLevel | null) => void;

export type SurfLevelButtonProps = {
  surfLevel: SurfLevel;
  selectedSurfLevel: SelectedSurfLevel;
  handleButtonPress: (input: SelectedSurfLevel) => void;
};
