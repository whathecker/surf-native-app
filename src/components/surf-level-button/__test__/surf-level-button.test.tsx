/* eslint-disable @typescript-eslint/no-unused-vars */
import * as React from "react";
import renderer from "react-test-renderer";
import SurfLevelButton from "../surf-level-button";
import { SelectedSurfLevel } from "../../../types/surf-profile";

const mockPressHandler = (_input: SelectedSurfLevel): void => {
  return;
};

describe("Test SurfLevelButton - inactive state", () => {
  it("should renders inactive surf level button for novice level correctly", () => {
    const tree = renderer
      .create(
        <SurfLevelButton
          handleButtonPress={mockPressHandler}
          selectedSurfLevel="beginner"
          surfLevel="novice"
        />,
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("should renders inactive surf level button for beginner level correctly", () => {
    const tree = renderer
      .create(
        <SurfLevelButton
          handleButtonPress={mockPressHandler}
          selectedSurfLevel="novice"
          surfLevel="beginner"
        />,
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("should renders inactive surf level button for intermediate level correctly", () => {
    const tree = renderer
      .create(
        <SurfLevelButton
          handleButtonPress={mockPressHandler}
          selectedSurfLevel="advanced"
          surfLevel="intermediate"
        />,
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("should renders inactive surf level button for advanced level correctly", () => {
    const tree = renderer
      .create(
        <SurfLevelButton
          handleButtonPress={mockPressHandler}
          selectedSurfLevel="intermediate"
          surfLevel="advanced"
        />,
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});

describe("Test SurfLevelButton - active state", () => {
  it("should renders active surf level button for novice level correctly", () => {
    const tree = renderer
      .create(
        <SurfLevelButton
          handleButtonPress={mockPressHandler}
          selectedSurfLevel="novice"
          surfLevel="novice"
        />,
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("should renders active surf level button for beginner level correctly", () => {
    const tree = renderer
      .create(
        <SurfLevelButton
          handleButtonPress={mockPressHandler}
          selectedSurfLevel="beginner"
          surfLevel="beginner"
        />,
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("should renders active surf level button for intermediate level correctly", () => {
    const tree = renderer
      .create(
        <SurfLevelButton
          handleButtonPress={mockPressHandler}
          selectedSurfLevel="intermediate"
          surfLevel="intermediate"
        />,
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("should renders active surf level button for advanced level correctly", () => {
    const tree = renderer
      .create(
        <SurfLevelButton
          handleButtonPress={mockPressHandler}
          selectedSurfLevel="advanced"
          surfLevel="advanced"
        />,
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
