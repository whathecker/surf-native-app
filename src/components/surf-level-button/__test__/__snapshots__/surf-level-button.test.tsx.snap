// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test SurfLevelButton - active state should renders active surf level button for advanced level correctly 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "white",
      "borderColor": "blue",
      "borderRadius": 12,
      "borderStyle": "solid",
      "borderWidth": 2,
      "flexDirection": "row",
      "justifyContent": "center",
      "opacity": 1,
      "paddingBottom": 18,
      "paddingTop": 18,
      "width": "100%",
    }
  }
>
  <Text
    style={
      Object {
        "color": "black",
        "fontSize": 16,
      }
    }
  >
    Advanced
  </Text>
</View>
`;

exports[`Test SurfLevelButton - active state should renders active surf level button for beginner level correctly 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "white",
      "borderColor": "blue",
      "borderRadius": 12,
      "borderStyle": "solid",
      "borderWidth": 2,
      "flexDirection": "row",
      "justifyContent": "center",
      "opacity": 1,
      "paddingBottom": 18,
      "paddingTop": 18,
      "width": "100%",
    }
  }
>
  <Text
    style={
      Object {
        "color": "black",
        "fontSize": 16,
      }
    }
  >
    Beginner
  </Text>
</View>
`;

exports[`Test SurfLevelButton - active state should renders active surf level button for intermediate level correctly 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "white",
      "borderColor": "blue",
      "borderRadius": 12,
      "borderStyle": "solid",
      "borderWidth": 2,
      "flexDirection": "row",
      "justifyContent": "center",
      "opacity": 1,
      "paddingBottom": 18,
      "paddingTop": 18,
      "width": "100%",
    }
  }
>
  <Text
    style={
      Object {
        "color": "black",
        "fontSize": 16,
      }
    }
  >
    Intermediate
  </Text>
</View>
`;

exports[`Test SurfLevelButton - active state should renders active surf level button for novice level correctly 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "white",
      "borderColor": "blue",
      "borderRadius": 12,
      "borderStyle": "solid",
      "borderWidth": 2,
      "flexDirection": "row",
      "justifyContent": "center",
      "opacity": 1,
      "paddingBottom": 18,
      "paddingTop": 18,
      "width": "100%",
    }
  }
>
  <Text
    style={
      Object {
        "color": "black",
        "fontSize": 16,
      }
    }
  >
    I've never surfed before
  </Text>
</View>
`;

exports[`Test SurfLevelButton - inactive state should renders inactive surf level button for advanced level correctly 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "white",
      "borderColor": "black",
      "borderRadius": 12,
      "borderStyle": "solid",
      "borderWidth": 1,
      "flexDirection": "row",
      "justifyContent": "center",
      "opacity": 1,
      "paddingBottom": 18,
      "paddingTop": 18,
      "width": "100%",
    }
  }
>
  <Text
    style={
      Object {
        "color": "black",
        "fontSize": 16,
      }
    }
  >
    Advanced
  </Text>
</View>
`;

exports[`Test SurfLevelButton - inactive state should renders inactive surf level button for beginner level correctly 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "white",
      "borderColor": "black",
      "borderRadius": 12,
      "borderStyle": "solid",
      "borderWidth": 1,
      "flexDirection": "row",
      "justifyContent": "center",
      "opacity": 1,
      "paddingBottom": 18,
      "paddingTop": 18,
      "width": "100%",
    }
  }
>
  <Text
    style={
      Object {
        "color": "black",
        "fontSize": 16,
      }
    }
  >
    Beginner
  </Text>
</View>
`;

exports[`Test SurfLevelButton - inactive state should renders inactive surf level button for intermediate level correctly 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "white",
      "borderColor": "black",
      "borderRadius": 12,
      "borderStyle": "solid",
      "borderWidth": 1,
      "flexDirection": "row",
      "justifyContent": "center",
      "opacity": 1,
      "paddingBottom": 18,
      "paddingTop": 18,
      "width": "100%",
    }
  }
>
  <Text
    style={
      Object {
        "color": "black",
        "fontSize": 16,
      }
    }
  >
    Intermediate
  </Text>
</View>
`;

exports[`Test SurfLevelButton - inactive state should renders inactive surf level button for novice level correctly 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "white",
      "borderColor": "black",
      "borderRadius": 12,
      "borderStyle": "solid",
      "borderWidth": 1,
      "flexDirection": "row",
      "justifyContent": "center",
      "opacity": 1,
      "paddingBottom": 18,
      "paddingTop": 18,
      "width": "100%",
    }
  }
>
  <Text
    style={
      Object {
        "color": "black",
        "fontSize": 16,
      }
    }
  >
    I've never surfed before
  </Text>
</View>
`;
